docs/src/main/asciidoc/spring-data-jpa-guide.adoc

```diff
@@ -0,0 +1,592 @@
+Moreover, developers know at build time whether or not their repository method names can be converted to proper JPQL queries.
+This also means that if a method name indicates that a field should be used that is not part of the Entity, developers will get
+the relevant error at build time.
+
+=== What is supported
+
+The following sections described the most important supported features of Spring Data JPA.
+
+==== Automatic repository implementation generation
+
+Interfaces that extend any of the following Spring Data repositories are automatically implemented:
+
+* org.springframework.data.repository.Repository
+* org.springframework.data.repository.CrudRepository
+* org.springframework.data.repository.PagingAndSortingRepository
+* org.springframework.data.jpa.repository.JpaRepository
+
+The generated repositories are also registered as beans so they can be injected into any other bean.
+Furthermore the methods that update the database are automatically annotated with `@Transactional`.
+
+==== Fine tuning of repository definition
+
+This allows user defined repository interfaces to cherry-pick methods from any of the supported Spring Data repository interfaces without having to extend those interface.
+This is particularly useful when for example a repository needs to use some methods from `CrudRepository` but it's undesirable to expose the full list of methods of said interface.
+
+Assume for example we have a `PersonRepository` that shouldn't extend `CrudRepository` but would like to use `save` and `findById` methods which are defined in said interface.
+In such a case we would simply create `PersonRepository` like so:
+
+[source,java]
+----
+package org.acme.spring.data.jpa;
+
+import org.springframework.data.repository.Repository;
+
+public class PersonRepository extends Repository<Person, Long> {
+
+    Person save(Person entity);
+
+    Optional<Person> findById(Person entity);
+}
+----
+
+==== Customizing individual repositories using repository fragments
+
+Repositories can be enriched with additional functionality or override the default implementation of methods of the supported Spring Data repositories.
+This is best shown with an example.
+
+A repository fragment is defined as so:
+
+[source,java]
+----
+public interface PersonFragment {
+
+    // custom findAll
+    List<Person> findAll();
+
+    void makeNameUpperCase(Person person);
+}
+----
+
+The implementation of that fragment looks like this:
+
+[source,java]
+----
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import io.quarkus.hibernate.orm.panache.runtime.JpaOperations;
+
+@Component
+public class PersonFragmentImpl implements PersonFragment {
+
+    @Override
+    public List<Person> findAll() {
+        // do something here
+        return (List<Person>) JpaOperations.findAll(Person.class).list();
```

Comment:
I find it a bit confusing you're using Panache here? Is it what you wanted to do?

Do we really need the cast btw?

Reply:
I wanted to showcase that fragment capability, which allows users to add custom functionality. If you would like me to do it using vanilla EntityManager, then I can certainly do it.
And yes, the cast seems to be necessary